blockchain/pool.go

```diff
@@ -556,17 +567,8 @@ OUTER_LOOP:
 			bpr.reset()
 			continue OUTER_LOOP // When peer is removed
 		case <-bpr.gotBlockCh:
-			// We got the block, now see if it's good.
-			select {
-			case <-bpr.pool.Quit():
-				bpr.Stop()
-				return
-			case <-bpr.Quit():
-				return
-			case <-bpr.redoCh:
-				bpr.reset()
-				continue OUTER_LOOP
-			}
+			bpr.Stop()
```

Comment:
Can't just stop here, <-bpr.redoCh can happen if the block received was bad (e.g. byzantine peer).

Reply:
Thus the comment, "now see if it's good.".  brp doesn't stop until it gets a good block.

Reply:
oh right. was hard to follow. we should make the comment clearer